libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/routefetch/RouteOptionsProvider.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.ui.app.internal.routefetch
+
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.extensions.applyDefaultNavigationOptions
+import com.mapbox.navigation.base.extensions.applyLanguageAndVoiceUnitOptions
+import com.mapbox.navigation.core.MapboxNavigation
+
+class RouteOptionsProvider {
+
+    private var adapter: (RouteOptions) -> RouteOptions = { it }
```

Comment:
Why not Builder to Builder? Now you'll get one extra `toBuilder` and 1 extra `build`.

Reply:
Mainly for consistency with `RerouteOptionsAdapter`, which serves a similar purpose. But we also have `CarRouteOptionsInterceptor` and other interfaces that operate on a builder like `TripNotificationInterceptor`, so why not do the same here. Updated.